src/test/shared/sam/cli/samCliLocalInvoke.test.ts

```diff
@@ -0,0 +1,329 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as del from 'del'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { writeFile } from '../../../../shared/filesystem'
+import { mkdtemp } from '../../../../shared/filesystemUtilities'
+import { TestLogger } from '../../../../shared/loggerUtils'
+import { DefaultSamCliTaskInvoker } from '../../../../shared/sam/cli/samCliInvoker'
+import { SamCliLocalInvokeInvocation } from '../../../../shared/sam/cli/samCliLocalInvoke'
+
+describe('SamCliLocalInvokeInvocation', async () => {
+
+    class FakeTaskExecution implements vscode.TaskExecution {
```

Comment:
Is there a reason that one class is called `Fake...`, and the other is called `Test...`?

Reply:
Fake is a plain stub, and Test is tailored to the needs of the tests. But not really